firmware/src/networking/protocol/mod.rs

```diff
@@ -0,0 +1,68 @@
+//! THIS MODULE IS WIP AND NOT FINISHED YET
+//!
+//! Manages the slimevr firmware protocol. All (de)serialization and state associated with
+//! the protocol happens in here.
+#![allow(dead_code)]
+
+mod serialize;
+
+use crate::{Quat, Unreliable};
+
+use embassy_sync::{
+	blocking_mutex::raw::NoopRawMutex,
+	channel::{Channel, Receiver, Sender},
+};
+use firmware_protocol::{CbPacket, SbPacket};
+
+pub struct ImuHandle {
```

Comment:
How will temporary sensor failures (`SbPacket::SensorInfo`) be communicated to protocol task? Will sensor "debounce" state be stored here, or in protocol? [Previously discussed on Discord](https://discord.com/channels/817184208525983775/1025861916805050409/1059745935149760543)

Reply:
I barely touched `ImuHandle` so far, haven't thought of it yet. We can add a function to signal failures, or use enums.

Reply:
Using another signal for failure state makes juggling the awaits in Protocol trickier (and goes against the principle of making invalid states unrepresentable), and [using enums is basically just repeating the Udp protocol](https://canary.discord.com/channels/817184208525983775/1025861916805050409/1059749464765640715)

Reply:
this code is WIP reeeeeeeeeeeeeeeeeeeeeeeeeeee